packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -135,57 +136,54 @@ export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
           const hasGetConfig = fileExportsGetConfig(filePath);
 
           if (filePath.endsWith('/_layout.tsx')) {
-            fileInfo.push({
-              type: 'layout',
-              path: filePath.replace('_layout.tsx', ''),
-              src,
-              hasGetConfig,
-            });
+            continue;
```

Comment:
Does this mean we no longer check types for `_/layout.tsx`?

Reply:
Did you mean "/_layout.tsx" btw?

Reply:
yeah `/_layout.tsx`. What I wonder is, previously, it's generated so if we mistype `render: 'sttatic'` (with `as const`), TS can catch it. But, isn't it no longer the case with this change?

Reply:
This would result in `render: 'dynamic'` for the type since that is the fallback for when render does not match static or dynamic.

So this would really just change the types if `staticPaths` was added in addition to the `sttatic` typo. But the behavior would be to just assume string for the slugs, and not error otherwise.

I think if we want to catch typos like that we should move to using `satisfies` probably

Reply:
I thought with `as const`, this typo can be detected. (If your concern is the TS/JS behavioral mismatch, we can add console.warn for JS, but technically, it's another story.)

Reply:
> I thought with as const, this typo can be detected.

How would you expect it to be detected?

> If your concern is the TS/JS behavioral mismatch

this ^ is not a concern here. Js is simply ignored because we are generating types for ts.

Reply:
because with `as const`, the `render` type is `'sttatic'` literal. Oh, I think you mean how I would notice it. Basically, CI or running tsc command.

Reply:
Gotcha! I was able to fix this so that it errors now :)

<img width="1090" alt="image" src="https://github.com/user-attachments/assets/788b2236-b8f6-4556-ab4b-664fbe76acbb">

Reply:
Hmm, I can't reproduce this with 28204476f3d2a119177ad8ce1f005b6eaa68caeb.

Reply:
hm, it still seems to work for me. Are you just making the typo in one of the `getConfig`'s in 01_template still?

Reply:
yes, `/_layout.tsx`.
here's the change:
```diff
diff --git a/examples/01_template/src/pages/_layout.tsx b/examples/01_template/src/pages/_layout.tsx
index 6d227c9f..bc360ef1 100644
--- a/examples/01_template/src/pages/_layout.tsx
+++ b/examples/01_template/src/pages/_layout.tsx
@@ -34,6 +34,6 @@ const getData = async () => {
 
 export const getConfig = async () => {
   return {
-    render: 'static',
+    render: 'sttatic',
   } as const;
 };
```

I get the JS runtime error:
```
Cannot process SSR Error: Invalid layout configuration
```
(I didn't know this is already implemented. 😓 )

Reply:
Ah! Yeah this wouldn't currently get checked because the generated types file only reads from pages, not layouts.

If we want to type check for scenarios like this I would recommend the `satisfies` or `GetConfig` approaches we've talked through.

Reply:
> I thought this continue was related to this change
Yes, the continue is where I am skipping `_layout.tsx` files for what gets written to pages.gen.ts

> my question was why it needs satisfies, because it seems to me as const can infer to a literal.
So satisfies is not necessary, you are correct that `as const` infers literal. 

So I think what you are leading towards is generating the import and typecheck for the layout files in `pages.gen.ts`. This is possible and I can change this to do that. My hesitance is that it seems like it would be better to check for typos where `getConfig` is defined and not in a separate generated file. 

If this is what you'd prefer though, that is fine, I'll change to add a check.

Reply:
> it would be better to check for typos where getConfig is defined and not in a separate generated file.

I agree it's better.

My original question was that this leads to capability regression with this PR, but as we already have the validity check in the JS runtime, I notice it's not super important. So, let's leave it. Thanks for the explanation.